security/security-spring/src/main/java/org/cbioportal/security/spring/authentication/keycloak/SAMLUserDetailsServiceImpl.java

```diff
@@ -107,8 +107,7 @@ public Object loadUserBySAML(SAMLCredential credential)
             if (attrName.equals(SAML_IDP_METADATA_ROLE_ATTR_NAME)) {
                 List<XMLObject> attributeValues = cAttribute.getAttributeValues();
                 if (!attributeValues.isEmpty()) {
-                    userRoles.add(new StringBuilder(APP_NAME).append(":").append(
-                        getAttributeValue(attributeValues.get(0))).toString());
+                    userRoles.add(getAttributeValue(attributeValues.get(0)));
```

Comment:
in theory Keycloak will only send the roles for the specific client application, so this change should be OK. However, we just noticed that Keycloak's SAML response contains all roles of **all** clients. This means that the use case of having multiple cbioportal applications using the same keycloak service will not work...Possible actions:
- find Keycloak documentation to configure Keycloak in such a way that  only the client roles are sent back (you can log a support request - see http://www.keycloak.org/community.html ). Adjust our documentation and/or code accordingly.
- if Keycloak solution is not available: log an issue at Keycloak to ensure only the client roles are sent back
- document alternative way / workaround, e.g. adding "cbioportal:" in front of every Keycloak role

Reply:
Independent of #3061, should `PUBLIC` studies still work? The user has not been given any authorities via Keycloak and I set
`always_show_study_group=PUBLIC` and imported `study_es_0`, but I don't see any studies in the portal.

This is relevant part of the log, I believe:
```
hasPermission(), checking permissions on targetDomainObject
getAppName(), APP_NAME = cbioportal
filterGroupsByAppName(), FILTER_GROUPS_BY_APP_NAME = false
PUBLIC_CANCER_STUDIES_GROUP= null
hasPermission(), cancer study stable id: study_es_0
hasPermission(), user: org.cbioportal.security.spring.authentication.PortalUserDetails@99cb6bf6: Username: xx@yy.de; Password: [PROTECTED]; Enabled: true; AccountNonExpired: true; credentialsNonExpired: true; AccountNonLocked: true; Not granted any authorities
```

Reply:
@messersc for PUBLIC to work, you have to make sure the `groups` column in `cancer_study` table also contains PUBLIC for the public studies. To automate this, you need to make sure the `groups` meta data field in used in your `meta_study.txt` file when loading a study into cBioPortal. See here for more info: https://github.com/cBioPortal/cbioportal/blob/master/docs/File-Formats.md#cancer-study and https://github.com/cBioPortal/cbioportal/blob/master/docs/User-Authorization.md#configuring-public-studies

Reply:
@messersc also remember when changing a portal.property like `always_show_study_group=PUBLIC`, you normally need to rebuild and redeploy the .war. Did you do this?

Reply:
@pieterlukasse I rebuild my docker image, yes. Also, `study_es_0` has `PUBLIC` in its groups, I just checked. So that should have been fine.

So I think this comment https://github.com/cBioPortal/cbioportal/pull/2525#issuecomment-326616201 is correct? Or would you expect different behaviour?